URBNSwiftyConvenience/Classes/LayoutConvenience.swift

```diff
@@ -135,3 +135,31 @@ public extension UIStackView {
         }
     }
 }
+
+public extension NSLayoutConstraint {
+    
+    public enum ConstraintPriority {
+        case low
+        case high
+        case required
+        case custom(priority: Float)
```

Comment:
I don’t think this is a bad idea, but in Swift4, UILayoutPriority _is_ a float backed enum so I see this only as a temporary solution.

Also, with the swift 4 version, someone [added an extension to the `UILayoutPriority` type](https://useyourloaf.com/blog/easier-swift-layout-priorities/) to add support for addition and subtraction. This can be useful if you need to make 1 view slightly higher than another:

```swift
let priority = UILayoutPriority.low + 1
```

At the very least, I’d like a TODO to remove this once swift4.

Reply:
also, why isn’t this just a float backed enum?